browser-test/src/end_to_end_enumerators.test.ts

```diff
@@ -1,54 +1,432 @@
   waitForPageJsLoad,
 } from './support'
+import {Page} from 'playwright'
 
 test.describe('End to end enumerator test', () => {
   const programName = 'Ete enumerator program'
-  const ctx = createTestContext(/* clearDb= */ false)
 
-  test('Updates enumerator elements in preview', async () => {
-    const {page, adminQuestions} = ctx
-    await loginAsAdmin(page)
+  test.describe('Admin page', () => {
+    test('Updates enumerator elements in preview', async ({
+      page,
+      adminQuestions,
+    }) => {
+      await loginAsAdmin(page)
+
+      await adminQuestions.gotoAdminQuestionsPage()
+
+      await page.click('#create-question-button')
+      await page.click('#create-enumerator-question')
+      await waitForPageJsLoad(page)
+
+      // Click the add button in the preview to ensure we get an entity row and corresponding delete
+      // button.
+      await page.click('button:text("Add Sample repeated entity type")')
+      // Validate that the field rendered
+      await validateScreenshot(page, 'enumerator-field')
+
+      // Now update the text when configuring the question and ensure that
+      // the preview values update.
+      await page.fill('text=Repeated Entity Type', 'New entity type')
+      await validateScreenshot(page, 'enumerator-type-set')
+
+      // Verify question preview has the default values.
+      await adminQuestions.expectCommonPreviewValues({
+        questionText: 'Sample question text',
+        questionHelpText: '',
+      })
+      await adminQuestions.expectEnumeratorPreviewValues({
+        entityNameInputLabelText: 'New entity type name #1',
+        addEntityButtonText: 'Add New entity type',
+        deleteEntityButtonText: 'Remove New entity type #1',
+      })
+    })
 
-    await adminQuestions.gotoAdminQuestionsPage()
+    test('Create nested enumerator and repeated questions as admin', async ({
+      page,
+      adminQuestions,
+      adminPrograms,
+    }) => {
+      await setupEnumeratorQuestion(
+        page,
+        adminQuestions,
+        adminPrograms,
+        /* shouldValidateScreenshot= */ true,
+      )
+    })
+  })
 
-    await page.click('#create-question-button')
-    await page.click('#create-enumerator-question')
-    await waitForPageJsLoad(page)
+  test.describe('Applicant flow with North star flag disabled', () => {
+    test.beforeEach(async ({page, adminQuestions, adminPrograms}) => {
+      await setupEnumeratorQuestion(
+        page,
+        adminQuestions,
+        adminPrograms,
+        /* shouldValidateScreenshot= */ false,
+      )
+    })
+
+    test('has no accessibility violations', async ({
+      page,
+      applicantQuestions,
+    }) => {
+      await applicantQuestions.applyProgram(programName)
+
+      await applicantQuestions.answerNameQuestion('Porky', 'Pig')
+      await applicantQuestions.clickNext()
+
+      // Check that we are on the enumerator page
+      expect(await page.isVisible('.cf-question-enumerator')).toEqual(true)
```

Comment:
**suggestion:** Related to conversation in eng meeting - can we change this to await `expect(` - here and below